src/bin/fusion/fusion.cpp

```diff
@@ -602,17 +611,75 @@ void jaiabot::apps::Fusion::loop()
+        latest_engineering_status.set_bot_id(latest_bot_status_.bot_id());
+        latest_engineering_status.mutable_rf_disable_options()->set_rf_disable_timeout_mins(
+            rf_disabled_timeout_mins_);
+
+        // If we get an engineering query for bot status
+        if (latest_engineering_status.query_bot_status())
         {
-            glog.is_debug1() && glog << "Publishing bot status over intervehicle(): "
+            glog.is_debug1() && glog << "Publishing queried bot status over intervehicle(): "
                                      << latest_bot_status_.ShortDebugString() << endl;
             intervehicle().publish<jaiabot::groups::bot_status>(latest_bot_status_);
-            last_bot_status_report_time_ = now;
+            latest_engineering_status.set_query_bot_status(false);
+        }
+
+        if ((last_bot_status_report_time_ + std::chrono::milliseconds(bot_status_rate_)) <= now)
+        {
+            // If bot_status_rate_ is not -1 then send bot status
+            if (bot_status_rate_ != -1)
+            {
+                glog.is_debug1() && glog << "Publishing bot status over intervehicle(): "
+                                         << latest_bot_status_.ShortDebugString() << endl;
+                intervehicle().publish<jaiabot::groups::bot_status>(latest_bot_status_);
+                last_bot_status_report_time_ = now;
+
+                // If the rf is disabled and operator enables rf
+                // then send powerstate command to enable WIFI
+                if (rf_disabled_)
+                {
+                    rf_disabled_ = false;
+                    latest_engineering_status.mutable_rf_disable_options()->set_rf_disable(
+                        rf_disabled_);
+                    // Send message to enable RF on PI (WIFI)
+                    interprocess().publish<jaiabot::groups::powerstate_command>(
+                        latest_engineering_status);
+                }
+            }
+            else
+            {
+                // If the rf is enabled and operator disables rf
+                // then send powerstate command to disable WIFI
+                if (!rf_disabled_)
+                {
+                    rf_disabled_ = true;
+                    latest_engineering_status.mutable_rf_disable_options()->set_rf_disable(
+                        rf_disabled_);
+                    // Send message to disable RF on PI (WIFI)
+                    interprocess().publish<jaiabot::groups::powerstate_command>(
+                        latest_engineering_status);
+                }
+
+                // If the rf_disable timeout has been reach then start sending bot status
+                if ((last_bot_status_report_time_ +
+                     std::chrono::minutes(rf_disabled_timeout_mins_)) <= now)
+                {
+                    // Set bot status rate to 1 Hz if the rf disable timeout is reached
+                    latest_engineering_status.set_bot_status_rate(
```

Comment:
We should use the bot status rate from the config file

Reply:
Added a todo... I need to add a function that reads in the enum and then sets the bot_status_period

I will create ticket.